src/components/Modals/PrintModal.vue

```diff
@@ -1,136 +1,108 @@
 <script setup>
-import { onMounted, ref, watch } from 'vue'
+import { ref } from 'vue'
 import { Dialog, DialogPanel, DialogTitle, TransitionChild, TransitionRoot } from '@headlessui/vue'
 import { PrinterIcon } from '@heroicons/vue/24/outline'
 import { ExclamationCircleIcon } from '@heroicons/vue/24/outline'
 import StandardButton from '@/components/Shared/StandardButton.vue'
 
 const props = defineProps({
-  showNotification: Boolean,
+  showPrintModal: Boolean,
   validQRCode: Boolean
 })
-const emit = defineEmits(['update:show-modal', 'print'])
-const updateLocalShowNotification = (value) => {
-  open.value = value
-  emit('update:show-modal', value)
-}
+const emit = defineEmits(['hideModal', 'print'])
 
-// INITIALISE TEMPLATE REFS
-const selectAll = ref(null)
-const intermediate = ref(null)
-const refs = ref([])
-onMounted(() => {
-  refs.value = Array(printOptions.length)
-    .fill(null)
-    .map(() => ref(null))
-})
-function setRef(index) {
-  return (el) => {
-    refs.value[index] = el
-  }
-}
+const disableButton = ref(false)
+const selectedOptions = ref(['code', 'name', 'email', 'org', 'role'])
 
-const notificationContent = ref({
-  titleText: '',
-  messageText: '',
-  buttonText: ''
-})
+const titleText = ref(props.validQRCode ? 'Select items to print' : 'Error!')
+const messageText = ref(props.validQRCode ? '' : 'Please scan a valid QR code')
```

Comment:
this shouldn't be used like this. refer to computed property

Reply:
these cannot be computed because they have to be reassigned a new value in the print function

Reply:
can be computed. computed trigger upon any value that is changed within the function
https://vuejs.org/guide/essentials/computed.html#basic-example